src/main/java/ch/njol/skript/expressions/ExprPortalCooldown.java

```diff
@@ -0,0 +1,128 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.SimplePropertyExpression;
+import ch.njol.skript.util.Timespan;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.GameMode;
+import org.bukkit.entity.Entity;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Portal Cooldown")
+@Description({
+	"The amount of time before an entity can use a portal. By default, it is 15 seconds after exiting a nether portal " +
+	"or end gateway. Players in survival/adventure get a cooldown of 0.5 seconds, while those in creative get no cooldown. " +
+	"Resetting will set the cooldown back to the default 15 seconds for non-player entities and 0.5 seconds for players."
+})
+@Examples({
+	"on portal:",
+		"\twait 1 tick",
+		"\tset portal cooldown of event-entity to 5 seconds"
+})
+@Since("INSERT VERSION")
+public class ExprPortalCooldown extends SimplePropertyExpression<Entity, Timespan> {
+
+	static {
+		register(ExprPortalCooldown.class, Timespan.class, "portal cooldown", "entities");
+	}
+
+	// Default cooldown for nether portals is 15 seconds:
+	// https://minecraft.fandom.com/wiki/Nether_portal#Behavior
+	private static final int DEFAULT_COOLDOWN = 15 * 20;
+	// Players only get a 0.5 second cooldown in survival/adventure:
+	private static final int DEFAULT_COOLDOWN_PLAYER = 10;
+
+	@Override
+	@Nullable
+	public Timespan convert(Entity entity) {
+		return Timespan.fromTicks(entity.getPortalCooldown());
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		switch (mode) {
+			case SET:
+			case ADD:
+			case RESET:
+			case DELETE:
+			case REMOVE:
+				return CollectionUtils.array(Timespan.class);
+			default:
+				return null;
+		}
+	}
+
+	@Override
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		Entity[] entities = getExpr().getArray(event);
+		int change = delta == null ? 0 : (int) ((Timespan) delta[0]).getTicks();
```

Comment:
I used getTicks because getTicks_i is being deprecated in #5307 and I thought it better to use the correct one now instead of fixing this again at a later time.

Reply:
@UnderscoreTud should I keep it or change it?

Reply:
I think you should use `getTicks_i()` for now. As that PR is targeted for 2.8 not 2.7